### ISSUE TITLE
JS errors when loading admin UI with custom Sync Function

### ISSUE THREAD

**Issue by jamiltz** [couchbaselabs/sync_gateway_admin_ui#20]
Using `log` and `_` in the sync function causes JS errors when loading the admin UI.

For the following:

```javascript
function(doc, oldDoc) {
  log(_.values(doc));
  log(_.values(oldDoc));
  if (oldDoc) {
    delete oldDoc._rev;
    delete doc._rev;
  }
  if (_.isEqual(doc, oldDoc)) {
    throw({forbidden: "There are no changes"});
  }
}
```

Output in chrome console:
```
Uncaught ReferenceError: log is not defined
```

and

```
Uncaught ReferenceError: _ is not defined
```

**Comment by snej** [comment]
What's "`_`"? I've never seen that used in a sync function before.

**Comment by jamiltz** [comment]
The sync function's context exposes the [underscorejs](http://underscorejs.org/) library and `_` is the root object.
In an email from @zgramana

> The SG script host has no additional JS libraries nor exposes root objects like JSON. It's far more limited than a browser JS runtime in that regard.

> You either only use what's in the provided Underscore.js library, or you'll have to include library code with the sync function.

> http://underscorejs.org

**Comment by jchris** [comment]
The version of underscore included in Otto is 1.4.4 so I will bring that one into the Dev UI context. Later we can talk about upgrading both to 1.8.x (current version).

**Comment by jchris** [comment]
Merged https://github.com/couchbase/sync_gateway/pull/1082
